app/controllers/api/cloud_volumes_controller.rb

```diff
@@ -34,7 +34,8 @@ def delete_resource_main_action(_type, cloud_volume, _data)
 
     def options
       if (id = params["id"])
-        render_update_resource_options(id)
+        action = params["option_action"] || "update"
+        send("render_#{action}_resource_options", id)
```

Comment:
I'm concerned here that someone might pass a strange option_action, which could bypass security in the future.   Not sure if it's better to whitelist attach and detach or just have separate calls out to render_attach_resource_options render_update_resource_options.

Reply:
This was something that slipped through from the other PR in the split, it will not be merged here so we can look into it more in the other PR https://github.com/ManageIQ/manageiq-api/pull/1141/files#diff-ff1fde5895f3426d678a7f5731cc08c27f3602cda6453a66677502b4731f5173R37-R38